just_audio/lib/just_audio.dart

```diff
@@ -1002,7 +1003,8 @@ class AudioPlayer {
     final currentIndex = this.currentIndex;
     final audioSource = _audioSource;
     final durationCompleter = Completer<Duration?>();
-    _platform = Future<AudioPlayerPlatform>(() async {
+
```

Comment:
Can you explain what this change does?

Reply:
I figured out that this function was the cause for one of the timer errors. The problem was gone after refactoring it. I wish I could give you a better "explanation".

Reply:
On the surface both ways of writing it seem equivalent, but I am guessing that your version is guaranteed to execute the synchronous portion of the block in the same event cycle.